nixos/modules/programs/mouse-actions.nix

```diff
@@ -0,0 +1,17 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.mouse-actions;
+in
+  {
+    options.programs.mouse-actions = {
+      enable = mkEnableOption (lib.mdDoc ''
+        Enable mouse-actions, which adds the user to `plugdev` and `input` groups.
```

Comment:
I don't think this is possible, because the module cannot by itself know what user it needs to add to these groups. You could either add it as an option (something like `programs.mouse-actions.users = [ ]`) or just ensure `groups.plugdev` exists like some other modules do (e.g. hackrf, minipro, ...)

Reply:
Right. I think a simple approach is to add a few udev rules and then include them like [glasgow](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/hardware/glasgow.nix). I've pushed my first attempt at that, it seems to be working.